### DISCUSSION TITLE
show path from git root (instead of dir name)

### DISCUSSION THREAD

**Discussion by savv** [ohmyzsh/ohmyzsh#11731] [Help] [answered]
Hi I'm incredibly happy with the default oh my zsh installation/theme. I would only like to change one thing - when I'm in a git repo I see: `➜ curdirname git:(branchname)`

For example, at the root directory, called `mygitrepo`, I currently see `➜ mygitrepo git:(branchname)`; if I was in a subdirectory gitrepo/src/foobar, I currently see `➜ foobar git:(branchname)`

Instead I would like to see the following, respectively: `➜ mygitrepo git:(branchname)` and `➜ mygitrepo/src/foobar git:(branchname)`.

What's the best way to achieve this?

**Comment by bencodesall** [comment]
I too have this question, although in a different theme. `Agnoster` shows the full path from `$HOME`. In a deeply nested area the dir path can cause wrap around which can be distracting.

Instead of

```
USERNAME@HOSTNAME > ~/source/github.com/some_dir/repo_root/deeper_down/the_repo/structure > BRANCH
```

... , I would love to see simply:
```
repo_root/deeper_down/the_repo/dir_structure > BRANCH
```
 @savv Have you found a solution for this that is comprehensible and fits within the `.oh-my-zsh` theming constructs?

**Reply by rwmitchell** [reply]
This is not a practical solution if you're working in a lot random repos, but if you've only got a handful, the bookmarks plugin can simulate what  you want....at least it does on my system.

**Comment by mcornella** [comment, marked as answer]
Something like this would work:
```sh
print_gitdir_or_basename() {
    if git rev-parse --is-inside-work-tree &> /dev/null; then
        echo ${"$(git rev-parse --show-prefix)"%%/}
    else
        echo ${PWD:t} # equivalent to `basename $PWD`
    fi
}

# Put the function in the prompt
PROMPT='$(print_gitdir_or_basename)'"<and any other customizations>"
```

**Reply by bencodesall** [reply]
Putting in .zshrc just broke the promt all together just had my username and not even a space after it.
